### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by byerc**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -0,0 +1,263 @@
+import { Activity, singleS3StreamOutput } from 'aries-data';
+import rp from 'request-promise';
+import { flattenUrlOptions } from './util.js';
+
+const apiUrl = 'https://secure.aha.io/api/v1/';
+
+export default class Aha extends Activity {
```
Comment (trong-le) [inline lib/index.js]: change `Aha` to `AhaSource`

File: lib/index.js [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+
+    @singleS3StreamOutput('json')
+    async onTask(activityTask, config, lastExecuted) {
```
Comment (trong-le) [inline lib/index.js]: change `lastExecuted` to `executionDate`

File: lib/index.js [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+      this.log.info(`Running with config: ${JSON.stringify(config)}`);
```
Comment (trong-le) [inline lib/index.js]: 4 space indentation from here till L15

File: lib/index.js [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+            },
+            uri: `${apiUrl}${endpoint}${uriOptions}`,
+            json: true
+        };
+
+        const response = await rp(options);
+
+        const totalPages = response.pagination ? response.pagination.total_pages : 0;
+        if(totalPages <= 1)
+          return response;
```
Comment (trong-le) [inline lib/index.js]: 4 space indentation

File: lib/index.js [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+
+        //Key of returned object. 2 are returned. Splice "pagination".
+        let key = Object.keys(response).splice(0,1).toString();
+
+        //Start at 2 because if pages are 0 or 1 then the data has been returned
+        for(let page = 2; page <= totalPages; page++) {
+          //Update URI for next page
```
Comment (trong-le) [inline lib/index.js]: 4 space indentation here down to L59

File: lib/index.js [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+            body: body,
+            uri: `${apiUrl}${endpoint}`,
+            json: true
+        };
+        const response = await rp(options);
+        return response;
+    }
+
+    async getProducts(config) {
+      const endpoint = 'products';
```
Comment (trong-le) [inline lib/index.js]: indentation

File: lib/index.js [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+      const response = await this.apiGetRequest({endpoint, undefined, config});
+      return response;
+    }
+
+    async getUsers(config) {
+      const endpoint = 'users';
```
Comment (trong-le) [inline lib/index.js]: indentation

File: lib/index.js [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+      const response = await this.apiGetRequest({endpoint, undefined, config});
+      return response;
+    }
+
+    async getFeatures(config) {
+      const endpoint = 'features';
```
Comment (trong-le) [inline lib/index.js]: indentation

**Comment by trong-le** [comment]
@byerc 
- [ ] Add configurations and descriptions to readme, along with a LICENSE file [Example here](https://github.com/aries-data/aries-activity-github-source)
- [ ] Add scripts to circle.yml file. [Example here](https://github.com/aries-data/generator-aries-activity/blob/master/generators/app/templates/common/circle.yml). Replace `<%= imageName %>` with `astronomerio/aha-source`
- [ ] Change all indentation that is 2 spaces to be 4 spaces.

File: README.md [hunk 9]
```diff
@@ -250,5 +250,61 @@ The requirement id is the number of the specified issue.
 "requirementId": "2"
 ```
 
+##Resonse
```
Comment (benjamingregory) [inline README.md]: Response*

**Comment by byerc** [comment]
@trong-le ready for re-review. please pass me.

File: lib/index.js [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+import rp from 'request-promise';
+import { flattenUrlOptions } from './util.js';
+
+const apiUrl = 'https://secure.aha.io/api/v1/';
+
+export default class AhaSource extends Activity {
+
+    @singleS3StreamOutput('json')
+    async onTask(activityTask, config, executionDate) {
+      this.log.info(`Running with config: ${JSON.stringify(config)}`);
```
Comment (trong-le) [inline lib/index.js]: space between async and this.log.info is only 2, needs to be 4. L11 - L17

File: README.md [hunk 11]
```diff
@@ -1,5 +1,310 @@
...
+"ideaPortalId": "2"
+```
+
+####Requirement ID
+The requirement id is the number of the specified issue.
+```
+"requirementId": "2"
+```
+
+##Resonse
```
Comment (trong-le) [inline README.md]: Typo

**Comment by byerc** [comment]
3rd times a charm @trong-le 

Sorry about that..

**Pull request merged**